src/rules/useIsnanRule.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * @license
+ * Copyright 2013 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../lint";
+
+/**
+ * Implementation of the use-isnan rule.
+ */
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING = "Found an invalid comparison for NaN: ";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new UseIsnanRuleWalker(sourceFile, this.getOptions()));
+    }
+}
+
+class UseIsnanRuleWalker extends Lint.RuleWalker {
+
```

Comment:
Line 33, but no matter, not important enough at all to make you revise for. Merged!